spec/models/user_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+RSpec.describe User, type: :model do
+  before do
+    FactoryBot.create(:first_user)
+  end
+
+  context 'when user data is valid' do
+    let!(:user) { FactoryBot.build(:second_user) }
+
+    it 'should be valid' do
+      expect(user).to be_valid
+    end
+
+    it 'should have proper role' do
+      expect(user.role).to eq(FactoryBot.attributes_for(:second_user)[:role])
```

Comment:
don't compare it with factoryBot instances.

Reply:
@kth-wtag apu, done.